packages/pluggableWidgets/switch-web/src/components/Switch.tsx

```diff
@@ -0,0 +1,46 @@
+import { MouseEvent, KeyboardEvent, createElement } from "react";
+import classNames from "classnames";
+import { Alert } from "@mendix/piw-utils-internal";
+import { SwitchContainerProps } from "../../typings/SwitchProps";
+
+export interface SwitchProps extends Pick<SwitchContainerProps, "id" | "class" | "style" | "tabIndex" | "deviceStyle"> {
+    onClick?: (event: MouseEvent<HTMLDivElement>) => void;
+    onKeyDown?: (event: KeyboardEvent<HTMLDivElement>) => void;
+    isChecked: boolean;
+    editable: boolean;
+    validation: string | undefined;
+}
+
+export default function Switch(props: SwitchProps) {
+    return (
+        <div className={classNames("widget-switch", props.class, props.deviceStyle)} style={props.style}>
```

Comment:
If we choose to do DOM changes, let's investigate whether we can simplify this structure.

Reply:
During refinement we decided to keep the same DOM